### PR TITLE
SDKS 2576 - WebAuthn Lost credentials

### PR DESCRIPTION
**Pull request by jeyanthanperiyasamy**
(https://bugster.forgerock.org/jira/browse/SDKS-2576)

https://forgerock.zendesk.com/agent/tickets/94902

### CODE REVIEW

File: forgerock-core/src/main/java/org/forgerock/android/auth/EncryptedSharedPreferences.kt [hunk 1]
```diff
@@ -67,22 +83,45 @@ class EncryptedPreferences {
...
                 fileName,
-                masterKeyAlias,
+                builder.build(),
                 EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
                 EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
             )
         }
+
+        // master key recovery code
+        private fun recoverFromDefaultMasterKey(
```
Comment (witrisna) [inline forgerock-core/src/main/java/org/forgerock/android/auth/EncryptedSharedPreferences.kt:104]: May provide javadoc, that this method is to extract data which encrypted with MasterKey, and delete the master key.

File: forgerock-core/src/main/java/org/forgerock/android/auth/EncryptedSharedPreferences.kt [hunk 2]
```diff
@@ -67,22 +83,45 @@ class EncryptedPreferences {
...
+            context: Context,
+            fileName: String = "secret_shared_prefs" + context.packageName
+        ): MutableMap<String, MutableSet<String>> {
+            val cache = mutableMapOf<String, MutableSet<String>>()
+            // creating all 1.0.5 alpha files
+            try {
+                val secureSharedPreferences =
+                    createPreferencesFile(context, fileName)
+                secureSharedPreferences.all.entries.map { it.key }.forEach { key ->
+                    secureSharedPreferences.getStringSet(key, emptySet())?.let {
```
Comment (witrisna) [inline forgerock-core/src/main/java/org/forgerock/android/auth/EncryptedSharedPreferences.kt:114]: Just to confirm, we previously set data as StringSet?

Comment (jeyanthanperiyasamy) [reply forgerock-core/src/main/java/org/forgerock/android/auth/EncryptedSharedPreferences.kt:114]: yes , this is how we set it before in the WebAuthnDataRepository

```       encryptedPreferences.edit().apply {
                map.forEach {
                    this.putStringSet(it.key, it.value)
                }
            }.apply()
```

**Review by spetrov — APPROVED** [review]
👍🏻

**Pull request merged**